clang/test/CodeGen/fake-use-opt-flags.cpp

```diff
@@ -0,0 +1,30 @@
+/// Check that we generate fake uses only when -fextend-lifetimes is set and we
+/// are not setting optnone, or when we have optimizations set to -Og and we have
+/// not passed -fno-extend-lifetimes.
+// RUN: %clang_cc1 -emit-llvm -disable-llvm-passes -O0 -fextend-lifetimes %s -o - | FileCheck %s
```

Comment:
Intention of this RUN line is that no fake uses are generated, but there are no test-lines for ensuring that -> add an implicit-fake-not? And the other check-only FileCheck commands.

Reply:
Normally I would use implicit-check-not, but in this case we're `CHECK-NEXT`ing from the start of the function to the very end, which equally guarantees that no fake uses have been generated.

Reply:
We just need two RUN lines, one with -fextended-lifetimes and one without. -O group is completely orthogonal and does not need to be tested